pandas/tests/arrays/test_boolean.py

```diff
@@ -251,6 +251,13 @@ def test_coerce_to_numpy_array():
         np.array(arr, dtype="bool")
 
 
+def test_to_boolean_array_from_strings():
+    result = BooleanArray._from_sequence_of_strings(["True", "False"])
```

Comment:
can you add in null values e.g. ``''`` and ``NaN`` i think would work (as strings).

Reply:
It looks like this specific method breaks when passed other kinds of strings. I _think_ (correct me if I'm wrong) the null strings get handled upstream by `read_csv` according to `na_values`.

I could add in a `None` if you think that makes sense, or add additional null strings to the `read_csv` test (could parametrize over different NA strings)?

Reply:
it should handle nulls as well (they will be converted in the Boolean constructor), if not, then this is broken too.

Reply:
Indeed, the csv parser already handles the different null string representations. So testing here just with `np.nan` or None is fine.